src/transformers/Utils/FieldField.php

```diff
@@ -28,7 +28,7 @@ protected function getTemplateOverrideData()
             'field_name' => $this->getFieldName(),
             'bundle' => $this->parent['id'],
             'label' => !empty($this->data['label']) ? $this->data['label'] : $this->data['id'],
-            'description' => !empty($this->parent['description']) ? $this->parent['description'] : '',
+            'description' => !empty($this->getDataValue('description', '')) ? $this->getDataValue('description', '') : $this->parent['description'],
```

Comment:
hi @solimanharkas 

the second argument for getDataValue takes the default value, if the key could not be found. (See the method declaration in BaseField) 

You should also check if 'description' is set in $this->parent, otherwise the code will show a warning.

Reply:
hi @stmh ,
thank you for your feedback, I used now a function to check the data from the description field.

Reply:
I think something ws lost in the translation. `$this->getDataValue` takes a 2nd argument, please have a look of the [definition of getDataValue](https://github.com/factorial-io/phab-entity-scaffolder/blob/abb5688481793583a5f08fa4cda6db0a52440ae5/src/transformers/Utils/FieldBase.php#L76) and use that 2nd argument in your code.

Reply:
hi @stmh 
sorry for that, I am not sure, if that what you mean. 
in theory I should not have to check if `$desc_field` is empty, as that what the function `getDataValue` did.
but I get always an empty string because of `$this->template[$key]` is an empty string in my test case.

Reply:
i was thinking about passing the `parent['description']` (if any) as default to `getDataValue()`

sth like

```
  'description' => $this->getDataValue('description', $this->parent['description'] ?? ''),
```

Reply:
yes, I tried something like this, but it is not working.
if we have description in one field and parent description for the others, it will return empty string for all other field instead of the parent description.  it is because `$this->template[$key]` on `getDataValue()` is an empty string in this use case.

Reply:
hmm, I see the problem.
Then i'd add a new function in FieldBase like `getDataValueOptionalFromParent` e.g. sth like this

```
    public function getDataValueOptionalFromParent($key, $default)
    {
        return $this->data[$key] ?? $this->parent[$key] ?? $this->template[$key] ?? $default;
    }
```

and later
```
  'description' => $this->getDataValueOptionalFromParent(('description', ''),
```

Reply:
thank you.
I implemented that and tested it as well and it is working fine.

Reply:
thanks soliman, 

sorry for being so nitpicking, but the entity scaffolder is already very complicated so I tried to have a clean solution. Thanks for your work, good job!